spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -285,8 +291,9 @@ private void saveStandardTaskDefinition(TaskDefinition taskDefinition) {
 	}
 
 	private String createComposedTaskDefinition(String graph) {
-		return String.format("%s --graph=\"%s\"",
-				taskConfigurationProperties.getComposedTaskRunnerName(), graph);
+		String composedTaskDefinition = StringUtils.hasText(this.dataFlowUri) ? "%s --graph=\"%s\" --dataFlowUri=%s" : "%s --graph=\"%s\"";
```

Comment:
I don't think we should set this here. It would be better to check the appDeploymentProperties and commandLineArgs for this key and if not specified add it